lib/proxy/request_id_middleware.rb

```diff
@@ -0,0 +1,14 @@
+module Proxy
+  class RequestIdMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      Thread.current.thread_variable_set(:request_id, env['HTTP_X_REQUEST_ID']) if env.has_key?('HTTP_X_REQUEST_ID')
+      to_ret = @app.call(env)
+      Thread.current.thread_variable_set(:request_id, nil)
```

Comment:
This seems to break logging of the request ID in a default WEBrick configuration, I can't see the request ID at all in the logs.

It appears from adding debug that the thread var's being cleared by this line _before_ the log formatter's called, so the thread variable is always nil. Commenting it out fixes the logging, but will obviously lead to data leaking between requests.